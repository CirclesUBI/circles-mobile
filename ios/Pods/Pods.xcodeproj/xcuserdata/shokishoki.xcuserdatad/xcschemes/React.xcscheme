<?xml version="1.0" encoding="UTF-8"?>
<Scheme
   LastUpgradeVersion = "0930"
   version = "1.3">
   <BuildAction
      parallelizeBuildables = "YES"
      buildImplicitDependencies = "YES">
      <BuildActionEntries>
         <BuildActionEntry
            buildForAnalyzing = "YES"
            buildForTesting = "YES"
            buildForRunning = "YES"
            buildForProfiling = "YES"
            buildForArchiving = "YES">
            <BuildableReference
               BuildableIdentifier = "primary"
               BlueprintIdentifier = "E1FE208787E57E5DCF17D5EA6C6B8065"
               BlueprintName = "React"
               ReferencedContainer = "container:Pods.xcodeproj"
               BuildableName = "libReact.a">
            </BuildableReference>
         </BuildActionEntry>
      </BuildActionEntries>
   </BuildAction>
   <TestAction
      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
      selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
      shouldUseLaunchSchemeArgsEnv = "YES"
      buildConfiguration = "Debug">
      <AdditionalOptions>
      </AdditionalOptions>
   </TestAction>
   <LaunchAction
      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
      selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
      launchStyle = "0"
      useCustomWorkingDirectory = "NO"
      ignoresPersistentStateOnLaunch = "NO"
      debugDocumentVersioning = "YES"
      debugServiceExtension = "internal"
      buildConfiguration = "Debug"
      allowLocationSimulation = "YES">
      <AdditionalOptions>
      </AdditionalOptions>
   </LaunchAction>
   <ProfileAction
      savedToolIdentifier = ""
      useCustomWorkingDirectory = "NO"
      debugDocumentVersioning = "YES"
      buildConfiguration = "Release"
      shouldUseLaunchSchemeArgsEnv = "YES">
   </ProfileAction>
   <AnalyzeAction
      buildConfiguration = "Debug">
   </AnalyzeAction>
   <ArchiveAction
      buildConfiguration = "Release"
      revealArchiveInOrganizer = "YES">
   </ArchiveAction>
</Scheme>
