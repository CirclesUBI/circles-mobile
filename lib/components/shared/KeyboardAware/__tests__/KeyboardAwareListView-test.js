import { ListView } from 'react-native'
import React from 'react'
import KeyboardAwareListView from '../KeyboardAwareListView'
import renderer from 'react-test-renderer'

describe('KeyboardAwareListView', () => {
  it('renders the KeyboardAwareListView', () => {
    const tree = renderer.create(
      <KeyboardAwareListView
        dataSource={{rowIdentities: []}}
        />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls onKeyboardAwareViewLayout from onLayout', () => {
    this.wrapper = global.shallow(
      <KeyboardAwareListView
        dataSource={{rowIdentities: []}}
        />
    )
    const instance = this.wrapper.instance()
    instance._onKeyboardAwareViewLayout = jest.fn()
    this.wrapper.find(ListView).props().onLayout({nativeEvent: true})
    expect(instance._onKeyboardAwareViewLayout).toHaveBeenCalled()
  })
  it('calls onKeyboardAwareViewScroll from onScroll', () => {
    const onScroll = jest.fn()
    this.wrapper = global.shallow(
      <KeyboardAwareListView
        dataSource={{rowIdentities: []}}
        onScroll={onScroll}
        />
    )
    const instance = this.wrapper.instance()
    instance._onKeyboardAwareViewScroll = jest.fn()
    this.wrapper.find(ListView).props().onScroll({nativeEvent: true})
    expect(instance._onKeyboardAwareViewScroll).toHaveBeenCalled()
    expect(onScroll).toHaveBeenCalled()
  })
  it('calls updateKeyboardAwareViewContentSize from onContentSizeChange', () => {
    this.wrapper = global.shallow(
      <KeyboardAwareListView
        dataSource={{rowIdentities: []}}
        />
    )
    const instance = this.wrapper.instance()
    instance._updateKeyboardAwareViewContentSize = jest.fn()
    this.wrapper.find(ListView).props().onContentSizeChange()
    expect(instance._updateKeyboardAwareViewContentSize).toHaveBeenCalled()
  })
})
