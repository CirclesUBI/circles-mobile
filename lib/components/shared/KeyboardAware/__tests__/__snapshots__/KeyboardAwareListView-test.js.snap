// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardAwareListView renders the KeyboardAwareListView 1`] = `
<RCTScrollView
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  dataSource={
    Object {
      "rowIdentities": Array [],
    }
  }
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onScroll={[Function]}
  opacity={1}
  renderScrollComponent={[Function]}
  scrollEventThrottle={200}
  scrollToBottomOnKBShow={false}
  scrollToInputAdditionalOffset={75}
  startScrolledToBottom={false}
>
  <View />
</RCTScrollView>
`;
